tests/Feature/FlightControllerTest.php

```diff
@@ -2,547 +2,570 @@
+        'arrival_time' => '2023-09-11T15:19'
+    ];
+    $this->json('PUT', action(UpdateFlightController::class, ['flight' => $flight]), $invalidOrigin)
+        ->assertStatus(422)
+        ->assertJsonValidationErrors('origin_city_id');
+
+
+    $invalidDestination = [
+        'airline_id' => 1,
+        'origin_city_id' => 1,
+        'destination_city_id' => 999,
+        'departure_time' => '2023-09-11T11:18',
+        'arrival_time' => '2023-09-11T15:19'
+    ];
+    $this->json('PUT', action(UpdateFlightController::class, ['flight' => $flight]), $invalidDestination)
+        ->assertStatus(422)
+        ->assertJsonValidationErrors('destination_city_id');
+
+
+    $invalidCities = [
+        'airline_id' => 1,
+        'origin_city_id' => 999,
+        'destination_city_id' => -4,
+        'departure_time' => '2023-09-11T11:18',
+        'arrival_time' => '2023-09-11T15:19'
+    ];
+    $this->json('PUT', action(UpdateFlightController::class, ['flight' => $flight]), $invalidCities)
+        ->assertStatus(422)
+        ->assertJsonValidationErrors('origin_city_id')
+        ->assertJsonValidationErrors('destination_city_id');
+});
+
+it('updates a flight with invalid airline', function (){
+    $cities = City::factory(2)->create();
+    $airline = Airline::factory()->create();
+
+    foreach ($cities as $city) {
+        $airline->cities()->attach($city->id);
     }
 
-    /** @test */
-    public function testUpdateAFlightWithValidParameters()
-    {
-        $cities = City::factory(2)->create();
-        $airline = Airline::factory()->create();
-
-        foreach ($cities as $city) {
-            $airline->cities()->attach($city->id);
-        }
-
-        $flightData = [
-            'airline_id' => 1,
-            'origin_city_id' => 1,
-            'destination_city_id' => 2,
-            'departure_time' => '2023-09-11T11:18',
-            'arrival_time' => '2023-09-11T15:19'
-        ];
-
-        $flight = Flight::factory()->create($flightData);
-
-        $this->json('PUT', action([FlightController::class, 'update'], ['flight' => $flight]), $flightData)
-            ->assertStatus(200)
-            ->assertJson($flightData);
+    $flightData = [
+        'airline_id' => 1,
+        'origin_city_id' => 1,
+        'destination_city_id' => 2,
+        'departure_time' => '2023-09-11T11:18',
+        'arrival_time' => '2023-09-11T15:19'
+    ];
+
+    $flight = Flight::factory()->create($flightData);
+
+    $invalidAirline = [
+        'airline_id' => -5,
+        'origin_city_id' => 1,
+        'destination_city_id' => 2,
+        'departure_time' => '2023-09-11T11:18',
+        'arrival_time' => '2023-09-11T15:19'
+    ];
+    $this->json('PUT', action(UpdateFlightController::class, ['flight' => $flight]), $invalidAirline)
+        ->assertStatus(422)
+        ->assertJsonValidationErrors('airline_id');
+});
+
+it('updates flight with departure time after arrival time', function(){
+    $cities = City::factory(2)->create();
```

Comment:
not for now, but just FYI, generally this tests would be one file per method (create, update, get, list, delete, etc) and they'll tend to share the setup in the factories, so we put them on the very top of the file, in a `beforeEach` function provided by PEST